src/cpp/rtps/transport/TCPTransportInterface.cpp

```diff
@@ -898,7 +898,12 @@ void TCPTransportInterface::perform_listen_operation(
     if (rtcp_message_manager)
     {
         channel = channel_weak.lock();
-        endpoint_to_locator(channel->remote_endpoint(), remote_locator);
+        asio::error_code ec;
+        endpoint_to_locator(channel->remote_endpoint(ec), remote_locator);
+        if (ec)
+        {
+            remote_locator.kind = LOCATOR_KIND_INVALID;
+        }
```

Comment:
I would have a new method `Locator remote_endpoint_to_locator(channel)` that performs all these steps.